store/store.go

```diff
@@ -1390,12 +1390,14 @@ var download = func(ctx context.Context, name, sha3_384, downloadURL string, use
 		}
 		if finalErr != nil {
 			if shouldRetryError(attempt, finalErr) {
+				logger.Debugf("Retrying because of: %s", finalErr)
```

Comment:
Yes, it makes sense to do it in shouldRetryError - it's used also for retrying in userinfo.go and auth.go.